scrapy/utils/python.py

```diff
@@ -273,6 +275,7 @@ def equal_attributes(obj1, obj2, attributes):
     return True
 
 
+@deprecated
```

Comment:
I don't think it works on classes? Have you tried to instantiate it?

Reply:
@wRAR  Yep, this is working with classes as inherently we are calling the __init__ function .

I tested it out and it gave me the following warning : 

UserWarning: Call to deprecated function WeakKeyCache. Use The following API is deprecated instead.

Reply:
Well, the message is inaccurate, “deprecated function WeakKeyCache”. We need to either make `@deprecated` smart enough to say ‘class’ when needed, or use a different approach. We’ve been using the latter so far, I’m not sure if there are potential unexpected side effects of using `@deprecated` after changing it to say “class” for classes.

Reply:
@Gallaecio What is the latter approach which has been used so far?

Reply:
> @Gallaecio @wRAR What is the latter approach which has been used so far?

Logging a warning manually in the `__init__` method, with `ScrapyDeprecationWarning` as warning class, and `stacklevel=2` (so that the warning is about the call to `__init__`, and not about the implementation line where we log the warning).

> Also which is the existing test function?

I did not get this, sorry. Could you elaborate?